### DISCUSSION TITLE
AuxKernel to get an average within a radius

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#21112] [Q&A General] [answered]
Is there a system in MOOSE to get an average value of the variable within a radius of the element. Say I have a variable V that gets solved and for every element, I want to know what the V_avg is where the V is taken from elements within radius R of the element center. I saw the user objects with the layered average which seems to be close to what I want but only gets the average in one direction I believe. Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

the layered average is indeed what you want to look for. 
i think this one might do what you want
https://mooseframework.inl.gov/source/userobject/NearestRadiusLayeredAverage.html

You should double check the code and or the docs to see what it does exactly. I remember being surprised that it didn’t take into account the quadrature or the volumes, I don’t remember. 

Guillaume

**Reply by maxnezdyur** [reply]
Yes, that looks pretty close to what I want. I might have to update part of it because it only allows an average over a single direction, according to the docs. Looking through the code more to fully understand it.

**Reply by GiudGiud** [reply]
so what kind of angular discretization do you need? like a uniform quadrature?
